contracts/axelarnet-gateway/src/state.rs

```diff
@@ -53,286 +30,297 @@ pub enum Error {
     MessageAlreadyExists(CrossChainId),
 }
 
+#[cw_serde]
+pub(crate) struct Config {
+    pub chain_name: ChainName,
+    pub router: Addr,
+}
+
+#[cw_serde]
+pub enum ExecutableMessage {
+    NotExecuted(Message),
+    Executed(Message),
+}
+
+impl ExecutableMessage {
+    pub fn msg(&self) -> &Message {
+        match self {
+            ExecutableMessage::NotExecuted(msg) | ExecutableMessage::Executed(msg) => msg,
+        }
+    }
+
+    pub fn msg_owned(self) -> Message {
+        match self {
+            ExecutableMessage::NotExecuted(msg) | ExecutableMessage::Executed(msg) => msg,
+        }
+    }
+}
+
 pub(crate) fn save_config(storage: &mut dyn Storage, value: &Config) -> Result<(), Error> {
-    CONFIG.save(storage, value).map_err(Error::from)
+    Ok(CONFIG.save(storage, value)?)
 }
 
 pub(crate) fn load_config(storage: &dyn Storage) -> Result<Config, Error> {
-    CONFIG
-        .may_load(storage)
-        .map_err(Error::from)?
-        .ok_or(Error::MissingConfig)
+    CONFIG.may_load(storage)?.ok_or(Error::MissingConfig)
 }
 
-pub(crate) fn save_sent_msg(
+pub(crate) fn save_unique_contract_call_msg(
     storage: &mut dyn Storage,
-    key: CrossChainId,
+    cc_id: &CrossChainId,
     msg: &Message,
 ) -> Result<(), Error> {
-    match SENT_MESSAGES.may_load(storage, key.clone())? {
-        Some(_) => Err(Error::MessageAlreadyExists(key)),
-        None => SENT_MESSAGES.save(storage, key, msg).map_err(Error::from),
+    // these cc IDs are generated by the gateway and should be unique.
+    // If there is a collision, there must be a bug in the logic of the caller.
```

Comment:
I think at this location it should be an error, because this function cannot control how the cc_id is generated. We could panic further up the stack instead.